test/portforward.go

```diff
@@ -0,0 +1,71 @@
+package test
+
+import (
+	"bufio"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"syscall"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+type PortForwardType struct {
+	LocalPort uint32
+	cmd       *exec.Cmd
+}
+
+func PortForward(pod corev1.Pod, remotePort uint32) (*PortForwardType, error) {
```

Comment:
Hmm. I tried that but for some reason it did not work. I didn't investigate the issue much but I was not able to connect to the Jaeger port. Maybe I can give it another try. This `PortForward` function is a little smarter than the one upstream because it chooses a free port by itself instead of trying a pre-defined port which might fail.